src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -162,72 +121,98 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 return;
             }
 
+            var declaration = (StatementSyntax)declarator.Parent.Parent;
```

Comment:
shoudl be declarationStatement.  "declaration" makes me think it has type VariableDeclaration.